components/sst_importer/src/import_mode.rs

```diff
@@ -100,11 +106,11 @@ struct ImportModeCFOptions {
 }
 
 impl ImportModeCFOptions {
-    fn new() -> ImportModeCFOptions {
-        ImportModeCFOptions {
-            block_cache_size: 4 << 30,
-            level0_stop_writes_trigger: 1 << 30,
-            level0_slowdown_writes_trigger: 1 << 30,
+    fn optimized_for_import_mode(&self) -> Self {
+        Self {
+            block_cache_size: self.block_cache_size.max(4 << 30),
```

Comment:
🤔 According to the original relevant PR #2875 we're supposed to take the `min` instead of `max` here, the rationale is "avoid OOM".

https://github.com/tikv/tikv/blob/22db38d3c321fa7ee78c213afd59d85f5cab85fd/src/config.rs#L165-L169

Reply:
As far as I can remember, there are a few things to consider here:
1. During the import process, sometimes there are a lot of files in L0
2. Since we enable the option `pin_l0_filter_and_index_blocks`, L0 files can consume a huge amount of memory
3. We can not change `pin_l0_filter_and_index_blocks` dynamically, so we reduce the block cache size instead, hopefully it will "avoid OOM"
4. Import mode is not designed for read performance at all

So I think the real problem here is the potential memory usage of a lot of L0 files, reducing the block cache size is just a workaround. It did work fine in those days, but we can review other options too.

Reply:
I think that was a conservative setting to avoid a previously seen OOM in import mode. I don't see concrete reason to set it, but we may need a bit of testing.

Reply:
Let's remove this setting and keep this master-only at the moment.